cirq-web/cirq_ts/src/bloch_sphere/components/axes.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Vector3, LineDashedMaterial, BufferGeometry, Line, Group} from 'three';
+
+interface Axis {
+  points: [Vector3, Vector3];
+  hexColor: string;
+  readonly lineWidth: number;
+}
+
+/**
+ * Generates the axes for the Bloch sphere. The radius
+ * (length of the axes) and color of each axis are configurable.
+ */
+export class Axes extends Group {
+  readonly radius: number;
+  readonly xAxisColor: string;
+  readonly yAxisColor: string;
+  readonly zAxisColor: string;
+
+  /**
+   * Class constructor.
+   * @param radius The radius of the circle the axes will belong to. This should be the same as the sphere.
+   * @param xAxisColor The color of the x axis as a string.
```

Comment:
I think the overall sentiment of the TS should be make things as easy to customize as possible so that working with it from the Python/HTML side is as easy as possible. I think the same question is valid for the number of meridians displayable on the Bloch sphere, the radius of the circle, etc. That being said, I agree and don't see a need for per-axis customization.

I think this is something that we should probably introduce in the next PR with the design fixes. I like the enum for color schemes too, we could have a dark mode and similar themes.

Reply:
Yeah, I really meant "are you specifying these parameters anywhere, or just using the defaults"? I strongly recommend a YAGNI approach when designing interfaces. It will make it simpler to understand how the feature is intended to be used. It also means that you don't need to test all of the ways that a user might pass you bad data ;-). It sounds like these should just be constants for the application/class rather than parameters.

Reply:
Ah ok I see now..changing